plugins/x86/x86_lifter.ml

```diff
@@ -1264,12 +1264,13 @@ module ToIR = struct
           assn reg32_t o_rdx (Bil.Ite (equal_v, edx_e, dst_hi_e))
         ]
       | Xadd(t, dst, src) ->
-        let tmp = tmp t in
-        Bil.Move (tmp, Bil.(op2e t dst + op2e t src))
+        let tmp_res, tmp_dst, tmp_src = tmp t, tmp t, tmp t in
+        Bil.Move (tmp_src, op2e t src)
```

Comment:
1. Do not use `Bil` data constructors to create BIL expressions and statements. The should be considered as private. We will make them private at some point of time.

2. Try to write code that looks like pseudocode, the C-like pseudocode. Basically someone without knowledge of OCaml should understand this code. I understand that it is hard in this particular case without rewriting everything from scratch, but sticking with using BIL EDSL and keeping to the most possible syntax (even if it impacts the performance) would be the right first steps. For example,

```ocaml
| Xadd (t, dst_op, src_op) -> 
   let tmp_src = tmp t in
   let tmp_dst = tmp t in
   let tmp_res = tmp t in 
   let src = op2e src_op in
   let dst = op2e dst_op in
   Bil.[
     tmp_src := src;
     tmp_dst := dst;
     tmp_res := dst + src;
     assn t src_op dst;
     assn t dst_op (var res);
  ] @ set_flag_add !!t (Bil.var tmp_dst) (Bil.var tmp_src) dst
```

This is the first approximation, but still is much more readable.